lib/customer.rb

```diff
@@ -0,0 +1,42 @@
+require 'csv'
+
+class Customer
+  attr_reader :id
+  attr_accessor :email, :address
+
+  def initialize(id, email, address)
+    @id = id
+    @email = email
+    @address = address
+  end
+
+  def self.all
+    all_customers = []
+    CSV.read('data/customers.csv').each do |customer_row|
+      id = customer_row[0].to_i
+      email = customer_row[1]
+      address={}
+      address[:street] = customer_row[2]
+      address[:city] = customer_row[3]
+      address[:state] = customer_row[4]
+      address[:zip] = customer_row[5]
```

Comment:
You can add these keys to address inline:

```suggestion
      address = {
        street: customer_row[2]
        city: customer_row[3]
        state: customer_row[4]
        zip: customer_row[5]
      }
```